src/services/client/enrollment.ts

```diff
@@ -1,7 +1,13 @@
+import dayjs from "dayjs";
 import EnrollmentData from "@/interfaces/enrollment";
 import Enrollment from "@/entities/Enrollment";
+import InvalidBirthDateError from "@/errors/InvalidBirthDate";
 
 export async function createNewEnrollment(enrollmentData: EnrollmentData) {
+  if(dayjs(enrollmentData.birthday).isAfter(dayjs(dayjs().subtract(18, "year")))) {
```

Comment:
Acho que essa logica acabou ficando um dificil de ler e também possui um bug. Acontece que o dayjs reconhece  `MM-DD-YYYY` como formato de input padrão, o que acabou fazendo com que, no objeto dayjs, os dias se invertessem com os meses. Sugeri uma mudança que resolve esse problema, e deixa a leitura da logica um pouco menos complicada:
```suggestion
  if(dayjs(value, "DD-MM-YYYY").isAfter(dayjs().subtract(18, "year"))) {
```
Isso deve resolver, sugiro testar os casos de borda essa minha sugestão e ver se está tudo funcionando ok :)